### PR TITLE
Added Vehicle Parts as Appliances

### PR DESCRIPTION
**Pull request by sonphantrung**
#### Summary

Features "Add FOODCO, Welding Rig, Chemlab, Cooking rig, MetalMaster Forge Buddy, ~~Kitchen Unit,~~ KitchenMaster"

#### Purpose of change

Resurrect #57862
Add Vanilla and Blaze Industries' vehicle parts as appliance, with some edits

#### Describe the solution

- Added vehicle parts as appliances, with some easy copy and paste

#### Describe alternatives you've considered

**NONE**

#### Testing

~~Edited using online VSCode Editor, not yet tested~~
Tested, all functions work

#### Additional context

My Linux laptop is gone now D8

### CODE REVIEW

File: data/mods/BlazeIndustries/blaze_furniture.json [hunk 1]
```diff
@@ -1,68 +0,0 @@
-[
```
Comment (Maleclypse) [inline data/mods/BlazeIndustries/blaze_furniture.json:1]: why are you deleting a file in an obsoleted mod?

**Comment by sonphantrung** [comment]
I'm not having much idea about what I was wrong about linting now

**Comment by Maleclypse** [comment]
Needs linting : data/mods/BlazeIndustries/furniture_and_terrain/blaze_construction_group.json
Please read doc/JSON_STYLE.md
Needs linting : data/mods/BlazeIndustries/furniture_and_terrain/blaze_construction.json
Please read doc/JSON_STYLE.md
Needs linting : data/mods/BlazeIndustries/furniture_and_terrain/blaze_appliances.json
Please read doc/JSON_STYLE.md

These files need to be linted.  You can take each file individually and paste it into https://dev.narc.ro/cataclysm/format.html hit lint and then copy it back into your editor and save it. Once you've done that with all three files linting will be complete.

File: data/mods/BlazeIndustries/furniture_and_terrain/blaze_appliances.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    "type": "vehicle_part",
+    "id": "ap_afs_kitchen_rig",
+    "name": { "str": "KitchenMaster cooking buddy" },
+    "description": "An all-in-one kitchen unit, chemistry lab, and food preparation area.  'e'xamine it to use its hotplate to heat up food or drinks.",
+    "item": "afs_kitchen_rig",
+    "location": "structure",
+    "flags": [ "CARGO", "OBSTACLE", "APPLIANCE", "FLAT_SURF" ],
+    "copy_from": "afs_kitchen_rig"
+  }
+]
```
Comment (Maleclypse) [inline data/mods/BlazeIndustries/furniture_and_terrain/blaze_appliances.json:32]:
```suggestion
]

```

File: data/mods/BlazeIndustries/furniture_and_terrain/blaze_construction.json [hunk 3]
```diff
@@ -1,41 +1,74 @@
...
+    "category": "FURN",
+    "required_skills": [ [ "electronics", 3 ], [ "mechanics", 3 ] ],
+    "time": 20,
+    "qualities": [ { "id": "SCREW", "level": 1 }, { "id": "HAMMER", "level": 3 }, { "id": "SAW_M", "level": 2 } ],
+    "components": [ [ [ "cable", 10 ] ], [ [ "amplifier", 3 ] ], [ [ "refrigerant_tank", 2 ] ], [ [ "afs_fridge", 1 ] ] ],
+    "pre_note": "Further modifies a converted fridge to function as a freezer.  You can 'e'xamine it afterwards to take it down for mounting.",
+    "pre_special": "check_empty",
+    "post_terrain": "f_afs_freezer_vehicle"
+  }
+]
```
Comment (Maleclypse) [inline data/mods/BlazeIndustries/furniture_and_terrain/blaze_construction.json:74]:
```suggestion
]

```

File: data/mods/BlazeIndustries/furniture_and_terrain/blaze_construction_group.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    "type": "construction_group",
+    "id": "convert_fridge_power_supply",
+    "name": "Convert Fridge Power Supply"
+  },
+  {
+    "type": "construction_group",
+    "id": "convert_vehicle_fridge_to_freezer",
+    "name": "Convert Vehicle Fridge to Freezer"
+  }
+]
```
Comment (Maleclypse) [inline data/mods/BlazeIndustries/furniture_and_terrain/blaze_construction_group.json:27]:
```suggestion
]

```

**Comment by Maleclypse** [comment]
Probably nevermind on what I said above.  It looks like you were missing an ending space which our linter can't catch on several of the files.  Let's see if this passes json validation now.

**Comment by anothersimulacrum** [comment]
I think we may want to examine what exactly these appliances provide and make appropriate non-all-in-one appliances based off of that. For instance, I think the kitchen unit is essentially just a worse oven.

**Comment by Maleclypse** [comment]
> I think we may want to examine what exactly these appliances provide and make appropriate non-all-in-one appliances based off of that. For instance, I think the kitchen unit is essentially just a worse oven.

RV kitchens don’t even normally come with stoves I believe. < https://www.katom.com/162-C39ELGLASSBK.html?gclid=Cj0KCQjw-pCVBhCFARIsAGMxhAcKniSaZkJVbnu6i1mdCG9g_adccKHVAudbrvHEYMXMh9Rkw33jQpgaAqhbEALw_wcB> this griddle feature looks common on the ones I’m seeing.

**Comment by anothersimulacrum** [comment]
Exactly - the kitchen unit provides a hotplate, and doesn't provide an oven, so it's a strictly worse oven. There's no reason for it to be an appliance in my eyes.

**Comment by sonphantrung** [comment]
So, does that mean, all of Blaze Industries' appliances are not suitable?

**Comment by anothersimulacrum** [comment]
I have no strong opinions about modded appliances, but I would say no.

**Comment by sonphantrung** [comment]
Linting seems problematic to me. After constant linting using that site, I see no difference

**Comment by Maleclypse** [comment]
> Linting seems problematic to me. After constant linting using that site, I see no difference

Linting does seem to be acting up.  I've requested help from more technical contributors.

**Comment by dseguin** [comment]
I fixed the line endings for you in https://github.com/sonphantrung/Cataclysm-DDA/pull/1.
All you would need to do is merge that PR into your branch.

**Comment by sonphantrung** [comment]
Should I be worried about the failed checks?

**Comment by sonphantrung** [comment]
Damn it, can't `copy-from` now

**Comment by sonphantrung** [comment]
What gone wrong with the build?

**Comment by aMegaSloth** [comment]
I think we need some additional cleaning up.

A few of the mod's construction items are being baked into the base game json files. I am pretty sure can be pulled out in to it's contained json in the mod.

**Comment by sonphantrung** [comment]
What exactly are the items?

**Comment by aMegaSloth** [comment]
You've made edits to:

data/json/construction.json
data/json/construction_group.json
data/json/furniture_and_terrain/appliances.json

Those can be migrated back into the mod so it can be self contained.

**Comment by sonphantrung** [comment]
So, it means, we copy the files into the mod folder?

**Comment by aMegaSloth** [comment]
Ohh, I see. So installing vehicle style applications to the static bases.

**Pull request merged**